File: samples/email-a-poem/src/main/java/io/quarkiverse/langchain4j/sample/EmailService.java
package io.quarkiverse.langchain4j.sample;

import jakarta.enterprise.context.ApplicationScoped;
import jakarta.inject.Inject;

import dev.langchain4j.agent.tool.Tool;
import io.quarkus.logging.Log;
import io.quarkus.mailer.Mail;
import io.quarkus.mailer.Mailer;

@ApplicationScoped
public class EmailService {

    @Inject
    Mailer mailer;

    @Tool("send the given content by email")
    public void sendAnEmail(String content) {
        Log.info("Sending an email: " + content);
        mailer.send(Mail.withText("clement@redhat.com", "A poem for you", content));
    }

}

File: samples/email-a-poem-kotlin/src/main/kotlin/io/quarkiverse/langchain4j/sample/EmailService.kt
package io.quarkiverse.langchain4j.sample

import dev.langchain4j.agent.tool.Tool
import io.quarkus.logging.Log
import io.quarkus.mailer.Mail
import io.quarkus.mailer.Mailer
import jakarta.enterprise.context.ApplicationScoped
import jakarta.inject.Inject

@ApplicationScoped
class EmailService(
    val mailer: Mailer
) {
    @Tool("send the given content by email")
    fun sendAnEmail(content: String) {
        Log.info("Sending an email: $content")
        mailer.send(Mail.withText("sendMeALetter@quarkus.io", "A poem for you", content))
    }
}
